xds/internal/client/client_watchers_rds.go

```diff
@@ -22,6 +22,24 @@ import (
 	"time"
 )
 
+// Int64Range is a range for header range match.
+type Int64Range struct {
+	Start int64 `json:"start"`
+	End   int64 `json:"end"`
+}
+
+// HeaderMatcher represents header matchers.
+type HeaderMatcher struct {
+	Name         string      `json:"name,omitempty"`
+	InvertMatch  *bool       `json:"invertMatch,omitempty"`
+	ExactMatch   *string     `json:"exactMatch,omitempty"`
+	RegexMatch   *string     `json:"regexMatch,omitempty"`
+	PrefixMatch  *string     `json:"prefixMatch,omitempty"`
+	SuffixMatch  *string     `json:"suffixMatch,omitempty"`
+	RangeMatch   *Int64Range `json:"rangeMatch,omitempty"`
+	PresentMatch *bool       `json:"presentMatch,omitempty"`
+}
+
```

Comment:
Why add this to this package instead of xdsrouting?

Reply:
The same structs will passed to xds_resolver, as part of RDS update. And they will be used to generate json service config.

Reply:
We would also need more structs here, if I'm not mistaken.

```
type Route struct {
  Name string
  Match RouteMatch
  Action RouteAction
}

type RouteMatch struct {
  Path PathMatch
  Headers []HeaderMatcher
}

...
```

Reply:
We will need more. But it doesn't look exactly like this (those are like balancer configs, while the structs here are 1:1 with the RDS resp proto message).

And also not necessary in this PR.